python-frontend/src/main/java/org/sonar/python/semantic/SymbolTableBuilder.java

```diff
@@ -398,7 +402,7 @@ private String resolveFullyQualifiedNameBasedOnRelativeImport(List<Token> dotted
       if (filePath == null || dottedPrefix.size() > filePath.size()) {
         return null;
       }
-      String resolvedPackageName = String.join("", filePath.subList(0, filePath.size() - dottedPrefix.size()));
+      String resolvedPackageName = String.join(".", filePath.subList(0, filePath.size() - dottedPrefix.size()));
```

Comment:
When we import something like `a.b.c.d` - filepath contains a,b,c,d . So we cannot merge `a.b.c` into `abc` as a package name, because it could not be resolved later on. There could be also a package with name `abc` - so this will collide.